core/src/main/java/org/radargun/stages/cache/test/AbstractQueryStage.java

```diff
@@ -0,0 +1,272 @@
+package org.radargun.stages.cache.test;
+
+import org.radargun.config.Converter;
+import org.radargun.config.DefinitionElement;
+import org.radargun.config.Property;
+import org.radargun.config.PropertyHelper;
+import org.radargun.stages.test.OperationLogic;
+import org.radargun.stages.test.TestStage;
+import org.radargun.traits.InjectTrait;
+import org.radargun.traits.Queryable;
+import org.radargun.utils.NumberConverter;
+import org.radargun.utils.ObjectConverter;
+import org.radargun.utils.ReflexiveConverters;
+
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Base class for stages which contain queries
+ */
+public abstract class AbstractQueryStage extends TestStage {
+
+    @Property(optional = false, doc = "Full class name of the object that should be queried. Mandatory.")
+    protected String queryObjectClass;
+
+    @Property(optional = false, doc = "Conditions used in the query", complexConverter = ConditionConverter.class)
+    protected List<Condition> conditions;
+
+    @Property(doc = "Use projection instead of returning full object. Default is without projection.")
+    protected String[] projection;
+
+    @Property(doc = "Use sorting order, in form [attribute[:(ASC|DESC)]][,attribute[:(ASC|DESC)]]*. " +
+            "Without specifying ASC or DESC the sort order defaults to ASC. Default is unordereded.",
+            converter = SortConverter.class)
+    protected List<SortElement> orderBy;
+
+    @Property(doc = "Offset in the results. Default is none.")
+    protected long offset = -1;
+
+    @Property(doc = "Maximum number of the results. Default is none.")
+    protected long limit = -1;
+
+    @Property(doc = "Check whether all slaves got the same result, and fail if not. Default is false.")
+    protected boolean checkSameResult = false;
```

Comment:
I think 'checkSameResult' and 'checkSameThreadResult' fields shouldn't be pulled up, as they're only used in QueryStage. This way XSD schema will include those properties in ContinuousQueryStage, however their usage will have no effect.